repo2docker/buildpacks/julia.py

```diff
@@ -98,15 +144,45 @@ def get_assemble_scripts(self):
         require = self.binder_path('REQUIRE')
         return super().get_assemble_scripts() + [(
             "${NB_USER}",
-            # Pre-compile all libraries if they've opted into it.
-            # `using {libraryname}` does the right thing
+            # Install and pre-compile all libraries if they've opted into it.
+            # In v0.6, Pkg.resolve() installs all the packages, but in v0.7+, we
+            # have to manually Pkg.add() each of them (since the REQUIRES file
+            # format is deprecated).
+            # The precompliation is done via `using {libraryname}`.
             r"""
-            cat "%(require)s" >> ${JULIA_PKGDIR}/v0.6/REQUIRE && \
             julia -e ' \
-               Pkg.resolve(); \
-               for pkg in keys(Pkg.Reqs.parse("%(require)s")) \
-                pkg != "julia" && eval(:(using $(Symbol(pkg)))) \
-               end \
+                require_file = "%(require)s" ;\
```

Comment:
I think this is now complex enough to be moved to a file of its own and then called from here. This lets us stop mixing Julia and Bash. See https://github.com/jupyter/repo2docker/blob/master/repo2docker/buildpacks/conda/__init__.py#L83 for how to add a script to the repo. Then this script (which should be in Julia) can be called from here.

Reply:
Awesome, great idea. I'm pushing up a commit now that just makes this change: splitting it out into its own file! :) I moved them into a new folder, `buildpacks/julia`.

Good suggestion! :)

-----

It seems to be working correctly, except that the line I added to `rm /tmp/install-repo-dependencies.jl` is failing:
```bash
rm: cannot remove '/tmp/install-repo-dependencies.jl': Operation not permitted
The command '/bin/sh -c julia /tmp/install-repo-dependencies.jl "REQUIRE" && rm /tmp/install-repo-dependencies.jl' returned a non-zero code: 1F
```

So now the tests are failing.

Any ideas? It seems like this file has the same permissions as `conda/environment.frozen.yml`, which is deleted successfully by the conda buildpack [here](https://github.com/jupyter/repo2docker/blob/master/repo2docker/buildpacks/conda/__init__.py#L73).

I see that the python buildpack [copies its files to /home/main](https://github.com/jupyter/repo2docker/blob/master/repo2docker/buildpacks/legacy/__init__.py#L35).. Should I do that instead?

Reply:
Here's where I do the deleting:
https://github.com/jupyter/repo2docker/pull/393/files#diff-fcf7bb456679b31cc83386be059cb3abR150